### DISCUSSION TITLE
Measure battery powering through USB

### DISCUSSION THREAD

**Discussion by wael5300** [rlogiacco/BatterySense#35] [Q&A] [answered]
Mr.Roberto Lo Giacco
How are you?
It's Wael .
I want to Measure the voltage on the battery that powers the Arduino through USB.
Would you mind helping me?

**Comment by rlogiacco** [comment, marked as answer]
![Solar-Power-Manager-application](https://user-images.githubusercontent.com/199473/121804849-a9001a80-cc48-11eb-9792-c08e1028a770.jpg)
This is the diagram you shared when you contacted me by email, so I'll refer to this.
In this configuration, you have no way to measure the battery capacity because you don't have any way to know the battery voltage.
The Arduino is powered by a regulated voltage which is kept stable at `5 V` by the battery module. This configuration is described in the README as [Lesser than 5V with voltage booster](https://github.com/rlogiacco/BatterySense#lesser-than-5v-with-voltage-booster).
To be in a position to measure the battery capacity, you need to have a connection between the _battery positive terminal_ to one of the analog pins in your Arduino.

How? Well, here is a quick procedure:
1. take out the battery and disconnect the battery module
2. pick a multimeter and put it in "continuity check" mode
3. start checking the pins, with one of the probes on the battery receptacle, positive terminal and the other...
  * avoid those marked as `USB`, `5V` or `GND`, those are _regulated_ so not good for the purpose
  * search for those market as `BAT`, if there is any...
If you can measure continuity between the battery positive terminal and the pin, then that pin can be marked as `BAT+`, which is what you need.

If you have no such pin, then you need to solder a piece of wire (it can be very thin, even AWG30) to the underside of the board, where the battery receptacle positive side is: that is going to be your `BAT+`. 
Because you are using a single cell LiIon and your Arduino seemed to be powered by `5V` you are in the easiest configuration possible... just check the examples and the [project readme](https://github.com/rlogiacco/BatterySense#lesser-than-5v-with-voltage-booster) for instructions.

If this solves your query, please mark it as _answered_.
